client/Contexts/ConfigFeatureFlag/ConfigFeatureFlag.spec.tsx

```diff
@@ -0,0 +1,382 @@
+          )
+        );
+        expect(
+          getByText(JSON.stringify(expectedStatusLoading))
+        ).toBeInTheDocument();
+        await apolloMockResponse();
+        expect(
+          getByText(JSON.stringify(expectedStatusError))
+        ).toBeInTheDocument();
+        expect(getByText(JSON.stringify(expectedClient))).toBeInTheDocument();
+        expect(
+          getByText(JSON.stringify(expectedFeatureFlags))
+        ).toBeInTheDocument();
+      });
+
+      it('triggers a re query of config data when `triggerRefetch` is invoked', async () => {
+        const expectedClientError = emptyClientResponse;
+        const expectedFeatureFlagsError = emptyFeatureFlagResponse;
+        const expectedStatusError = errorStatus;
+
+        const expectedClientComplete = returnClientResponse;
+        const expectedFeatureFlagsComplete = featureFlagsNs(
+          mockFeatureFlagResponse
+        );
+        const expectedStatusComplete = successStatus;
+
+        const { getByText } = render(
+          withApolloProviderReturning(
+            [...errorResponse, ...successResponse],
+            <ConfigFeatureFlagProvider>
+              <ConfigFeatureFlagConsumer>
+                {(values) => {
+                  const { triggerRefetch } = values;
+                  return (
+                    <React.Fragment>
+                      {testRenderFunc(values)}
+                      <button onClick={triggerRefetch}>Refetch</button>
+                    </React.Fragment>
+                  );
+                }}
+              </ConfigFeatureFlagConsumer>
+            </ConfigFeatureFlagProvider>
+          )
+        );
+        // 1st tick - error state
+        await apolloMockResponse();
+        expect(
+          getByText(JSON.stringify(expectedStatusError))
+        ).toBeInTheDocument();
+        expect(
+          getByText(JSON.stringify(expectedClientError))
+        ).toBeInTheDocument();
+        expect(
+          getByText(JSON.stringify(expectedFeatureFlagsError))
+        ).toBeInTheDocument();
+        // trigger the refetch
+        userEvent.click(getByText('Refetch'));
+        // 2nd tick - refetch should have completed
+        await apolloMockResponse();
+        expect(
+          getByText(JSON.stringify(expectedStatusComplete))
+        ).toBeInTheDocument();
+        expect(
+          getByText(JSON.stringify(expectedClientComplete))
+        ).toBeInTheDocument();
+        expect(
+          getByText(JSON.stringify(expectedFeatureFlagsComplete))
+        ).toBeInTheDocument();
+      });
+    });
+  });
+
+  describe('`useConfigFeatureFlag` hook', () => {
+    const HookWrapper: FunctionComponent = () => {
+      const { triggerRefetch, ...others } = useConfigFeatureFlag();
+      return (
+        <p>
+          {JSON.stringify({
+            ...others,
+            triggerRefetch: typeof triggerRefetch === 'function',
+          })}
+        </p>
+      );
+    };
```

Comment:
Is there a reason you're doing it this way instead of using https://www.npmjs.com/package/@testing-library/react-hooks?

Reply:
Did not realise this was a thing! Will update to use it